### PR TITLE
Coverage for: factory/bootstrap

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Reasoning behind the pull request
- increase code coverage
  
## Proposed changes
- increase coverage on factory/bootstrap

## Testing procedure
- standard system test

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: factory/bootstrap/bootstrapComponents.go [hunk 1]
```diff
@@ -82,9 +82,6 @@ func NewBootstrapComponentsFactory(args BootstrapComponentsFactoryArgs) (*bootst
 	if args.WorkingDir == "" {
 		return nil, errors.ErrInvalidWorkingDir
 	}
-	if check.IfNil(args.StatusCoreComponents) {
```
Comment (iulianpascalau) [inline factory/bootstrap/bootstrapComponents.go:85]: :+1:

File: factory/bootstrap/bootstrapComponents_test.go [hunk 2]
```diff
@@ -1,116 +1,248 @@
...
 	t.Parallel()
 
-	args := componentsMock.GetBootStrapFactoryArgs()
-	args.WorkingDir = ""
-
-	bcf, err := bootstrap.NewBootstrapComponentsFactory(args)
-
-	require.Nil(t, bcf)
-	require.Equal(t, errorsErd.ErrInvalidWorkingDir, err)
+	// TODO: add tests for createLatestStorageDataProvider and createUnitOpener failure
```
Comment (iulianpascalau) [inline factory/bootstrap/bootstrapComponents_test.go:105]: TODO for a follow-up PR?

File: factory/bootstrap/shardingFactory.go [hunk 3]
```diff
@@ -29,6 +30,15 @@ func CreateShardCoordinator(
 	prefsConfig config.PreferencesConfig,
 	log logger.Logger,
 ) (sharding.Coordinator, core.NodeType, error) {
+	if check.IfNil(nodesConfig) {
+		return nil, "", errErd.ErrNilGenesisNodesSetupHandler
+	}
+	if pubKey == nil {
```
Comment (iulianpascalau) [inline factory/bootstrap/shardingFactory.go:36]: please use check.IfNil

**Review by iulianpascalau — APPROVED** [review]

File: errors/errors.go [hunk 4]
```diff
@@ -541,3 +541,9 @@ var ErrNilPersistentHandler = errors.New("nil persistent handler")
 
 // ErrNilGenesisNodesSetupHandler signals that a nil genesis nodes setup handler has been provided
 var ErrNilGenesisNodesSetupHandler = errors.New("nil genesis nodes setup handler")
+
+// ErrNilLogger signals that a nil logger instance has been provided
+var ErrNilLogger = errors.New("nil logger")
+
+// ErrNilShuffleOutCloser signals that a nil shuffler out closer has been provided
+var ErrNilShuffleOutCloser = errors.New("nil shuffler out closer")
```
Comment (ssd04) [inline errors/errors.go:549]: should we go with `shuffler out` or with `shuffle out` ?

File: factory/bootstrap/shardingFactory_test.go [hunk 5]
```diff
@@ -0,0 +1,661 @@
...
+			make(chan endProcess.ArgEndProcess, 1),
+			&nodeTypeProviderMock.NodeTypeProviderStub{},
+			&testscommon.EnableEpochsHandlerStub{},
+			&validatorInfoCacherMocks.ValidatorInfoCacherStub{},
+		)
+		require.NotNil(t, err)
+		require.True(t, check.IfNil(nodesC))
+	})
+	t.Run("NewIndexHashedNodesCoordinator fails should error", func(t *testing.T) {
+		t.Parallel()
```
Comment (ssd04) [inline factory/bootstrap/shardingFactory_test.go:490]: this test name and the previous one have the same name, rename here?

File: errors/errors.go (hunk 4 above)
Comment (sstanculeanu) [reply errors/errors.go]: updated, typo

File: factory/bootstrap/shardingFactory_test.go (hunk 5 above)
Comment (sstanculeanu) [reply factory/bootstrap/shardingFactory_test.go]: fixed

**Review by gabi-vuls — APPROVED** [review]
System test passed 
@@                            Log scanner                                  @@
##                            unittests_factory_bootstrap                  ##
================================================================================
- Known Warnings              11
- New Warnings                2
- Known Errors                0
- New Errors                  0
- Panics                      0
================================================================================
- block hash does not match                  2754
- wrong nonce in block                       912
- miniblocks does not match                  0
- num miniblocks does not match              0
- miniblock hash does not match              0
- block bodies does not match                0
- receipts hash missmatch                    0
================================================================================
-  No jailed nodes on the testnet
================================================================================

**Review by ssd04 — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]

**Pull request merged**